posthog/temporal/batch_exports/backfill_batch_export.py

```diff
@@ -163,7 +172,13 @@ async def backfill_schedule(inputs: BackfillScheduleInputs) -> None:
     full_backfill_range = backfill_range(start_at, end_at, frequency * inputs.buffer_limit)
 
     for backfill_start_at, backfill_end_at in full_backfill_range:
-        utcnow = dt.datetime.now(dt.timezone.utc)
+        utcnow = get_utcnow()
+
+        if end_at is None and backfill_end_at >= utcnow:
```

Comment:
We will also want some kind of detection in case we are too slow to catch-up with real time. For example: keeping track of the difference `utcnow - end_at` and ensure it keeps going down.

If it doesn't, we could be too slow to ever catch-up, and maybe we should give up early.

Reply:
Could be for later though, initially we'll be keeping an eye on the http migrations ourselves.

Reply:
For now these are manually monitored HTTP migrations (since we manually start them, and need to update the customer once they are realtime anyway).

I'm trying to think of how automated monitoring would actually work. So I track `utcnow - end_at` but I need to "put" that somewhere and observe it over time (i.e. I can't blow up the job right when it starts just because it appears slow). Would I store this somewhere in Temporal so I can see its value over time and only make a judgement after a few hours of data? Or what?

Reply:
We could use the last N durations to make a simple linear estimation of how much time is remaining in the backfill. Could be a useful metric to report to users too.

Reply:
Still, early cancellation would come down to defining a threshold of how much delay are we willing to tolerate before failing, and it may vary as the load on the cluster varies through the day, so I am not sure we could set a threshold that works honestly... Maybe keep it as a non-actionable metric (just for reporting) first?